eclipsesource/theia-cloud
# Support multiple operator replicas with leader election #201

* create test setup with multiple replicas
* implement leader election in operator
* add logging
* make leader election arguments configurable

Helm Chart changes: https://github.com/eclipsesource/theia-cloud-helm/pull/35

**How to test**
* Checkout this branch and the branch from the linked theia-cloud-helm PR
* Use the `test-configurations` to start up the `2-01_try-now` configuration
* Preload demo image using `minikube image load theiacloud/theia-cloud-demo:0.8.1.OSWeek23-v1`
* Start `minikube dashboard`
* Verify that there are two operator pods
* Verify that only one of the pods is the leader by checking the log messages
* Start up a session using the landing page. Verify that only the leader did some work. 
* Kill the leader pod. Verify that a new pod is launched to replace it. Verify that still only one pod is the leader. 

Fixes #201
